src/plugins/intel_gpu/tests/test_cases/gather_gpu_test.cpp

```diff
@@ -8,10 +8,334 @@
 #include <intel_gpu/primitives/gather.hpp>
 
 #include <cstddef>
+#include <array>
 
 using namespace cldnn;
 using namespace ::tests;
 
+std::ostream& operator<<(std::ostream& os, FLOAT16 x) {
+    return os << float(x);
+}
+
+template <class T>
+int dim(const T& a) {
+    int ret = a.size();
+    while (ret - 1 >= 0 && a[ret - 1] == 1)
+        ret--;
+    return ret;
+};
+
+template <class T>
+int mult(const T& a) {
```

Comment:
I think `multiplication` is a misleading name. Please use proper one. We have similar function for `struct layout`: `get_linear_size`

Reply:
Done.